### PR TITLE
fix audits and user management with uaa and no database

### LINKED ISSUE #9992: User Management admin pages not generated for gateway with no database
##### **Overview of the issue**

The user management pages are not generated in my gateway microservice.
This seems to have been introduced in version 6.0.1

##### **Suggest a Fix**

If I switch to use a database with my gateway instead on no database, it seems to work.

##### **JHipster Version(s)**

```
gateway@0.0.0 /tmp4/gateway
└── (empty)

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "databaseType": "no",
    "devDatabaseType": "no",
    "enableHibernateCache": false,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": [
      "en",
      "da"
    ],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "dk.sware.simplecrm.gateway",
    "packageFolder": "dk/sware/simplecrm/gateway",
    "prodDatabaseType": "no",
    "searchEngine": false,
    "serviceDiscoveryType": "eureka",
    "skipClient": false,
    "skipServer": false,
    "testFrameworks": [],
    "websocket": false,
    "baseName": "gateway",
    "applicationType": "gateway",
    "authenticationType": "uaa",
    "uaaBaseName": "uaa",
    "cacheProvider": "no",
    "buildTool": "gradle",
    "jhipsterVersion": "6.1.2",
    "skipUserManagement": false,
    "clientPackageManager": "npm",
    "serverPort": "8080",
    "clientFramework": "angularX",
    "useSass": true,
    "clientTheme": "none",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": []
  },
  "entities": [
    "Foo"
  ]
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Foo {
  bar String
}
microservice Foo with backend

</pre>
</details>

##### **Environment and Tools**

openjdk version "12.0.1" 2019-04-16
OpenJDK Runtime Environment (build 12.0.1+12)
OpenJDK 64-Bit Server VM (build 12.0.1+12, mixed mode, sharing)

git version 2.22.0

node: v10.16.0

npm: 6.9.0

yeoman: 2.0.6

yarn: 1.17.0

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by clement26695**
Should fix #9992 

However, I am wondering: it seems that we both rely on the same skipUserManagement in front end and backend, when with uaa, we want to generate userManagement for the front end, but not for back end. Should I refactor this logic instead of this PR ? 

Should I add test to see if expected files are correctly generated with the no database option ?

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/admin/admin.module.ts.ejs [hunk 1]
```diff
@@ -27,7 +27,7 @@ import { <%=angularXAppName%>SharedModule } from 'app/shared';
 
 import {
     adminState,
-    <%_ if (databaseType !== 'no' && databaseType !== 'cassandra') { _%>
+    <%_ if ((databaseType !== 'no' ? true : authenticationType === 'uaa') && databaseType !== 'cassandra') { _%>
```
Comment (cbornet) [inline generators/client/templates/angular/src/main/webapp/app/admin/admin.module.ts.ejs]: I think that's just `(databaseType !== 'no'  || authenticationType === 'uaa')`

Comment (clement26695) [reply generators/client/templates/angular/src/main/webapp/app/admin/admin.module.ts.ejs]: Indeed!

File: generators/client/templates/angular/src/main/webapp/app/admin/admin.module.ts.ejs [hunk 2]
```diff
@@ -57,7 +57,7 @@ import {
         RouterModule.forChild(adminState)
     ],
     declarations: [
-        <%_ if (databaseType !== 'no' && databaseType !== 'cassandra') { _%>
+        <%_ if ((databaseType !== || authenticationType === 'uaa') && databaseType !== 'cassandra') { _%>
```
Comment (wmarques) [inline generators/client/templates/angular/src/main/webapp/app/admin/admin.module.ts.ejs]: I don't understand the `!== ||` part... Does that work ?

Comment (clement26695) [reply generators/client/templates/angular/src/main/webapp/app/admin/admin.module.ts.ejs]: No, I forgot the 'no'...

**Comment by pascalgrimaud** [comment]
Reviewed by Christophe and William. 
It looks good.
I think we can stay like this, for now

**Pull request merged**